deepchecks/vision/batch_wrapper.py

```diff
@@ -94,6 +96,23 @@ def __len__(self):
         dataset = self._context.get_data_by_kind(self._dataset_kind)
         return len(list(dataset.data_loader.batch_sampler)[self.batch_index])
 
+    def vision_properties(self, raw_data: List, properties_list: List[Dict], input_type=PropertiesInputType):
+        """Calculate and cache the properties for the batch according to the propety input type"""
+        properties_list = validate_properties(properties_list)
+        # if there are no cached properties at all, calculate all the properties on the list,
+        # else calculate only those that were not yet calculated.
+        dataset = self._context.get_data_by_kind(self._dataset_kind)
+        if self._vision_properties_cache[input_type.value] is None:
+            self._vision_properties_cache[input_type.value] = dataset.calc_properties(raw_data, properties_list)
+        else:
+            properties_to_calc = [p for p in properties_list if p['name'] not in
+                                  self._vision_properties_cache[input_type.value].keys()]
+            self._vision_properties_cache[input_type.value].update(
+                                                                dataset.calc_properties(raw_data, properties_to_calc))
```

Comment:
What happens if we want to run them on a sample, and not on all of the data? Seems the cache is not per index of the data, right?

Reply:
@nirhutnik we decided together with @ItayGabbay that we cache on all the data and filter later, you can see ImagePropertyDrift for example